### ISSUE TITLE
Feature Tests Failing on Windows

### ISSUE THREAD

**Issue by gloryCU** [cucumber/cucumber-js#1852]
**Describe the bug**
While pairing with @mattwynne to set up my development environment, we found that the scenarios all failed. Same error: 
```
293) Scenario: custom world constructor is passed the parameters # features\world_parameters.feature:76
     × Before # features\support\hooks.ts:20
         Error: EPERM: operation not permitted, symlink 'C:\Users\gugwu\Documents\OSContribution\cucumber-js' -> 'C:\Users\gugwu\Documents\OSContribution\cucumber-js\tmp\world_parameters.feature_76\node_modules\@cucumber\cucumber'
             at Object.symlinkSync (fs.js:1210:3)
             at Object.createSymlinkSync (C:\Users\gugwu\Documents\OSContribution\cucumber-js\node_modules\fs-extra\lib\ensure\symlink.js:76:13)             at World.<anonymous> (C:\Users\gugwu\Documents\OSContribution\cucumber-js\features\support\hooks.ts:42:11)
             at Object.run (C:\Users\gugwu\Documents\OSContribution\cucumber-js\src\user_code_runner.ts:37:21)
             at Object.run (C:\Users\gugwu\Documents\OSContribution\cucumber-js\src\runtime\step_runner.ts:50:41)
             at processTicksAndRejections (internal/process/task_queues.js:95:5)
             at async TestCaseRunner.invokeStep (C:\Users\gugwu\Documents\OSContribution\cucumber-js\src\runtime\test_case_runner.ts:130:12)    
             at async TestCaseRunner.runHook (C:\Users\gugwu\Documents\OSContribution\cucumber-js\src\runtime\test_case_runner.ts:246:12)       
             at async C:\Users\gugwu\Documents\OSContribution\cucumber-js\src\runtime\test_case_runner.ts:201:20
             at async TestCaseRunner.aroundTestStep (C:\Users\gugwu\Documents\OSContribution\cucumber-js\src\runtime\test_case_runner.ts:162:28)     - Given a file named "features/passing_steps.feature" with: # features\step_definitions\file_steps.ts:10
         """
         Feature: a feature
           Scenario: a scenario
             Given the world parameters are correct
         """
     - Given a file named "features/support/world.js" with: # features\step_definitions\file_steps.ts:10
         """
         const {setWorldConstructor} = require('@cucumber/cucumber')    

         function CustomWorld(options) {
           for(const key in options.parameters) {
             this[key] = options.parameters[key]
           }
         }

         setWorldConstructor(CustomWorld)
         """
     - Given a file named "features/step_definitions/my_steps.js" with: 
# features\step_definitions\file_steps.ts:10
         """
         const assert = require('assert')
         const {Given} = require('@cucumber/cucumber')

         Given(/^the world parameters are correct$/, function() {       
           assert.equal(this.a, 1)
         })
         """
     - When I run cucumber-js with `--world-parameters '{"a":1}'` # features\step_definitions\cli_steps.ts:19
     - Then scenario "a scenario" step "Given the world parameters are correct" has status "passed" # features\step_definitions\message_steps.ts:68
     √ After # features\support\hooks.ts:104
```

**To Reproduce**
1. Open the repo on a Windows machine
2. Open the git bash prompt
3. Run `npm install`
4. Run `npm test`

**Expected behavior**
I expected all the tests to pass.

**Desktop (please complete the following information):**
 - : Windows 10 build 19042

**Comment by mattwynne** [comment]
It looks like maybe this has something to do with the Windows filesystem not supporting symlinks: https://github.com/npm/npm/issues/5824

Is there anyone else using Node and Windows who could advise us?

**Comment by leider** [comment]
tl;dr: Test run successfully in win10 enterprise (20H2), other findings with windows below.

Just read on twitter, was curious, so I tried. My findings:
1. npm install failed, probably b/o old node version `v14.18.1` -> solution; delete pckage-lock.json, then npm i installs in power shell
1. npm test fails early when running from power shell, b/o `cp src/importer.js lib/ ` in `build-local` target -> solution: use git bash
1. npm test in git bash succeeds

```
271 scenarios (271 passed)
1396 steps (1396 passed)
1m23.575s (executing steps: 2080m17.514s)
```

**Comment by davidjgoss** [comment]
Have you tried this in a command prompt with administrator privileges? Been a while since I’ve worked on windows but this rings a bell.

**Comment by leider** [comment]
Good point (the admin advice). My user has local admin rights. I dind't open any shell as Administrator though.

**Comment by aslakhellesoy** [comment]
FWIW, the CI build is passing on windows: https://github.com/cucumber/cucumber-js/actions/runs/1454067375

**Comment by aurelien-reeves** [comment]
From a fresh clone of the repo, I've been able to run `npm install-test` using git bash without special tweaking and without specific rights.

Which version of Windows are you using? Do you have specific policy?

**Comment by mattwynne** [comment]
@RukenaG let's close this when we get a chance to pair again and can test it works for you now.

**Comment by davidjgoss** [comment]
Closing as this has been quiet for a long time, but please comment again if this doesn't feel resolved.
